data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfiguration.java

```diff
@@ -121,24 +147,37 @@ public RestHighLevelClient createClient() {
       i++;
     }
     final RestClientBuilder restClientBuilder = RestClient.builder(httpHosts);
-    final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
-    if (username != null) {
-      LOG.info("Using the username provided in the config.");
-      credentialsProvider.setCredentials(
-              AuthScope.ANY, new UsernamePasswordCredentials(username, password));
-    }
-    final SSLContext sslContext = certPath != null ? getCAStrategy(certPath) : getTrustAllStrategy();
-    restClientBuilder.setHttpClientConfigCallback(
-            httpClientBuilder -> {
-              httpClientBuilder
-                      .setDefaultCredentialsProvider(credentialsProvider)
-                      .setSSLContext(sslContext);
-              if (this.insecure) {
-                httpClientBuilder.setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE);
-              }
-              return httpClientBuilder;
-            }
-    );
+    if(awsSigv4) {
+      //if aws signing is enabled we will add AWSRequestSigningApacheInterceptor interceptor,
+      //if not follow regular credentials process
+      LOG.info("{} is set, will sign requests using AWSRequestSigningApacheInterceptor", AWS_SIGV4);
```

Comment:
nit: can parse the two credential paths into a function for cleanliness. not a blocker at all